### PR TITLE
feat: use renovate/yarn as base image

### PR DESCRIPTION
**Pull request by viceice**
This pr uses `renovate/yarn` as base image.

So i removed node, npm and yarn.
I also adapted the multistage build to match the slim image

### CODE REVIEW

**Comment by ViceIce** [comment]
relates to #4708

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,38 @@
...
+ENV DEBIAN_FRONTEND noninteractive
+ENV LC_ALL C.UTF-8
+ENV LANG C.UTF-8
+
+WORKDIR /usr/src/app/
+
+# required for install
+USER root
+
+RUN chown -R ubuntu:ubuntu /usr/src/app/
```
Comment (rarkins) [inline Dockerfile]: I'm not sure this achieves anything here, unless we built renovate/yarn badly?

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,38 @@
...
+#============
+FROM renovate/yarn:1.19.1@sha256:9f034a5962dc0ad3212934346f179503cc29045f2551f03a359637bd79151134 AS base
+
+LABEL maintainer="Rhys Arkins <rhys@arkins.net>"
+LABEL name="renovate"
+LABEL org.opencontainers.image.source="https://github.com/renovatebot/renovate"
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV LC_ALL C.UTF-8
+ENV LANG C.UTF-8
```
Comment (rarkins) [inline Dockerfile]: These should already be set in our base image

File: Dockerfile [hunk 3]
```diff
@@ -101,7 +92,8 @@ ENV PATH $PATH:/opt/elixir-${ELIXIR_VERSION}/bin
 
 # PHP Composer
 
-RUN apt-get update && apt-get install -y php-cli php-mbstring && apt-get clean
+RUN apt-get update && apt-get install -y php-cli php-mbstring && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (rarkins) [inline Dockerfile]: What if we make this rm -rf lists improvement first, to reduce the changes to this PR?

**Review by rarkins — CHANGES_REQUESTED** [review]
I don't think our CI actually tests *building* so we need to test this manually too

File: Dockerfile (hunk 3 above)
Comment (ViceIce) [reply Dockerfile]: I can do this, this rm is simply best practice for docker with debian based distros

File: Dockerfile (hunk 1 above)
Comment (ViceIce) [reply Dockerfile]: maybe we no longer need to write to this path?

File: Dockerfile (hunk 3 above)
Comment (ViceIce) [reply Dockerfile]: see #4711

**Review by viceice — COMMENTED** [review]
Have tested it locally against gitlab with docker and yarn manger

File: Dockerfile (hunk 1 above)
Comment (rarkins) [reply Dockerfile]: OK let's remove this section and see if everything is still fine

**Comment by rarkins** [comment]
Eventually - maybe v21 - I want to stop bundling npm, yarn and pnpm with Renovate (i.e. remove them as npm dependencies). Until then, we are installing npm and yarn as part of "yarn install" anyway so I'm not sure if we're saving much versus using renovate/node as our base image for publishing instead?

**Comment by ViceIce** [comment]
We do not save space for now, but we save some build time and both images have the same base.

And after this it is mode easy to migrate to node v12, because we only need to update base image

**Comment by rarkins** [comment]
How would it be worse though if the final image was based off renovate/node instead of renovate/yarn? renovate/node should be a subset of renovate/yarn, and we don't run any `yarn x` commands anyway.

**Comment by ViceIce** [comment]
🤔 We can use the `renovate/node` for final image, but for build we need yarn.

**Comment by ViceIce** [comment]
If we use different base images we have to make sure, that both have the same node version.

also yarn only add 2mb (compressed) to the image

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,26 @@
...
+WORKDIR /usr/src/app/
+
+
+# Build image
+#============
+FROM base as tsbuild
+
+USER root
+
+# Python 2 and make are required to build node-re2
```
Comment (rarkins) [inline Dockerfile]: Shouldn't we be on python 3 now?

**Pull request merged**